src/hooks/browser-headers.ts

```diff
@@ -37,16 +38,27 @@ export async function browserHeadersHook(options: Options): Promise<void> {
         useHeaderGenerator,
         headerGenerator,
         proxyUrl,
-        sessionToken,
     } = context as Context;
+    const sessionData = context.sessionData as HeadersData | undefined;
 
     if (!useHeaderGenerator || !headerGenerator) return;
 
+    const create = () => ({
+        1: headerGenerator.getHeaders({
+            httpVersion: '1',
+            ...headerGeneratorOptions,
+        }),
+        2: headerGenerator.getHeaders({
+            httpVersion: '2',
+            ...headerGeneratorOptions,
+        }),
+    });
+
     const url = options.url as URL;
 
     const resolveProtocol = (options as any).resolveProtocol
         || (proxyUrl
-            ? createResolveProtocol(proxyUrl)
+            ? createResolveProtocol(proxyUrl, sessionData as any)
             : http2.auto.resolveProtocol);
```

Comment:
I think this `const resolveProtocol ... ` would deserve a refactor. It's getting more and more complex and it has nested conditions. Perhaps a separate function?